kong/api/init.lua

```diff
@@ -7,125 +7,142 @@ local api_helpers = require "kong.api.api_helpers"
 local tablex = require "pl.tablex"
 
 local find = string.find
-
 local app = lapis.Application()
-
 local needs_body = tablex.readonly({ PUT = 1, POST = 2, PATCH = 3 })
 
+
 local function parse_params(fn)
+
   return app_helpers.json_params(function(self, ...)
+
     if needs_body[ngx.req.get_method()] then
+
       local content_type = self.req.headers["content-type"]
       if content_type and find(content_type:lower(), "application/json", nil, true) then
         if not self.json then
           return responses.send_HTTP_BAD_REQUEST("Cannot parse JSON body")
         end
       end
+
     end
+
     self.params = api_helpers.normalize_nested_params(self.params)
     return fn(self, ...)
   end)
+
 end
 
+
 local function on_error(self)
   local err = self.errors[1]
+
   if type(err) == "table" then
     if err.db then
       return responses.send_HTTP_INTERNAL_SERVER_ERROR(err.message)
+
     elseif err.unique then
       return responses.send_HTTP_CONFLICT(err.tbl)
+
     elseif err.foreign then
       return responses.send_HTTP_NOT_FOUND(err.tbl)
+
     else
       return responses.send_HTTP_BAD_REQUEST(err.tbl or err.message)
     end
   end
+
 end
 
+
 app.default_route = function(self)
   local path = self.req.parsed_url.path:match("^(.*)/$")
 
   if path and self.app.router:resolve(path, self) then
     return
+
   elseif self.app.router:resolve(self.req.parsed_url.path.."/", self) then
     return
   end
 
   return self.app.handle_404(self)
 end
 
+
 app.handle_404 = function(self)
   return responses.send_HTTP_NOT_FOUND()
 end
 
+
 app.handle_error = function(self, err, trace)
-  ngx.log(ngx.ERR, err.."\n"..trace)
-  -- We just logged the error so no need to give it to responses and log it twice
-  return responses.send_HTTP_INTERNAL_SERVER_ERROR()
+
+  if err and err:match("don't know how to respond to") then
```

Comment:
`string.match` should probably just perform a plain text search if not patterns are needed here.